internal/internal_event_handlers.go

```diff
@@ -871,6 +874,32 @@ func (wc *workflowEnvironmentImpl) GetRegistry() *registry {
 	return wc.registry
 }
 
+// ResetLAWFTAttemptCounts resets the number of attempts in this WFT for all LAs to 0 - should be
+// called at the beginning of every WFT
+func (wc *workflowEnvironmentImpl) ResetLAWFTAttemptCounts() {
```

Comment:
Can we write an integration test to cover this function as well? I think `TestSendsCorrectMeteringData` would still pass even if this function wasn't called.

Reply:
It fails if the call is removed - otherwise the attempts which span a heartbeat are over-counted.